dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/manually-set-incomplete/manually-set-incomplete.event-handler.ts

```diff
@@ -1,12 +1,18 @@
-import { UtilisationReportEntity } from '@ukef/dtfs2-common';
-import { NotImplementedError } from '../../../../../errors';
+import { EntityManager } from 'typeorm';
+import { DbRequestSource, UtilisationReportEntity } from '@ukef/dtfs2-common';
 import { BaseUtilisationReportEvent } from '../../event/base-utilisation-report.event';
 
-export type UtilisationReportManuallySetIncompleteEvent = BaseUtilisationReportEvent<'MANUALLY_SET_INCOMPLETE', undefined>;
+type ManuallySetIncompleteEventPayload = {
+  requestSource: DbRequestSource;
+  transactionEntityManager: EntityManager;
+};
+
+export type UtilisationReportManuallySetIncompleteEvent = BaseUtilisationReportEvent<'MANUALLY_SET_INCOMPLETE', ManuallySetIncompleteEventPayload>;
 
-export const handleUtilisationReportManuallySetIncompleteEvent = (
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const handleUtilisationReportManuallySetIncompleteEvent = async (
   report: UtilisationReportEntity,
+  { requestSource, transactionEntityManager }: ManuallySetIncompleteEventPayload,
 ): Promise<UtilisationReportEntity> => {
-  throw new NotImplementedError('TODO FN-1862');
+  report.setAsNotCompleted({ requestSource });
+  return await transactionEntityManager.getRepository(UtilisationReportEntity).save(report);
```

Comment:
Hmm is there a way we can call our custom repo methods within the context of the transaction? If so could we use that repo here so we aren't calling `getRepository` directly other than in repos and in tests?

Reply:
Unfortunately not. I'm going to leave a long reply comment below regarding the options we have with respect to transactions in typeorm, but the "query runner" approach requires you to use the entity manager supplied by the query runner instance - if we used `UtilisationReportRepo` here, the query would run outside the context of the transaction